independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InterceptorCreator.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.arc;
+
+import jakarta.interceptor.InvocationContext;
+
+/**
+ * This interface is used by synthetic interceptor to initialize an interceptor instance.
+ */
+public interface InterceptorCreator {
+
+    /**
+     *
+     * @param context
+     * @return the intercept function
+     */
+    InterceptFunction create(SyntheticCreationalContext<Object> context);
+
+    @FunctionalInterface
+    public interface InterceptFunction {
+
+        /**
+         *
+         * @param invocationContext
+         * @return the return value
+         * @throws Exception
+         */
+        Object intercept(InvocationContext invocationContext) throws Exception;
```

Comment:
I'd use `ArcInvocationContext` as a parameter here just to give users access to additional functionality we have there.

Reply:
Good point. I think I'm +1 to that.

Reply:
Done.